vortex-array/src/array/chunked/compute/filter.rs

```diff
@@ -106,24 +102,26 @@ fn filter_slices<'a>(
             }
         } else {
             // start != end means that the range is split over at least two chunks:
-            // start chunk: append a slice from (start_idx, start_chunk_end).
+            // start chunk: append a slice from (start_idx, start_chunk_end), i.e. whole chunk.
             // end chunk: append a slice from (0, end_idx).
             // chunks between start and end: append ChunkFilter::All.
-            let start_chunk_end = chunk_ends[start_chunk + 1];
-            let start_slice = (start_idx, start_chunk_end as _);
+            let start_chunk_len = chunk_ends[start_chunk + 1] - chunk_ends[start_chunk];
+            let start_slice = (start_idx, start_chunk_len as _);
             match &mut chunk_filters[start_chunk] {
                 f @ (ChunkFilter::All | ChunkFilter::None) => {
                     *f = ChunkFilter::Slices(vec![start_slice])
                 }
                 ChunkFilter::Slices(slices) => slices.push(start_slice),
             }
 
-            let end_slice = (0, end_idx);
-            match &mut chunk_filters[end_chunk] {
-                f @ (ChunkFilter::All | ChunkFilter::None) => {
-                    *f = ChunkFilter::Slices(vec![end_slice]);
+            if end_idx != 0 {
```

Comment:
Now the edge case that comes up is end_idx can be 0 if you have a single element chunk that you want to take the element from since end_idx is exclusive